### PR TITLE
Create sctgen tool

### PR DESCRIPTION
**Pull request by RJPercival**
Makes it easier to create SCTs for test purposes. This tool carries some risk, since creating an SCT using a CT log's prod key would be an easy way to accidentally violate its MMD, but I've included such a warning in the README which hopefully mitigates that.

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [x] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

**Comment by Martin2112** [comment]
What do you think about removing the private key option? If they're purely test keys they arguably don't need to be password protected and this could prevent an error being made.

File: ctutil/sctgen/sctgen.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	logID, err := ctfe.GetCTLogID(logSigner.Public())
+	if err != nil {
+		glog.Exitf("Error calculating log ID: %v", err)
+	}
+
+	timestamp, err := time.Parse(time.RFC3339, *timestampStr)
+	if err != nil {
+		glog.Exitf("Error parsing --timestamp: %v", err)
+	}
+	timestampMillis := uint64(timestamp.UnixNano() / 10e6)
```
Comment (kroeckx) [inline ctutil/sctgen/sctgen.go]: You want 1e6, not 10e6

**Comment by AlCutter** [comment]
Ah, CLA-bot is unhappy that you left :(

**Pull request closed without merging**